xarray/namedarray/pycompat.py

```diff
@@ -102,6 +102,10 @@ def to_numpy(
     from xarray.core.indexing import ExplicitlyIndexed
     from xarray.namedarray.parallelcompat import get_chunked_array_type
 
+    if hasattr(data, "to_numpy"):
```

Comment:
this is the ugly bit. It's only to test that we are using the `to_numpy` method instead of `np.asarray` which would fail for cupy.

Reply:
@keewis does this seem OK to you?